sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -609,7 +629,8 @@ object ScalaJSPluginInternal {
         val libs =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
         resolvedJSEnv.value match {
-          case env: LinkingUnitJSEnv =>
+          case env: LinkingUnitJSEnv
+              if scalaJSModuleKind.value == ModuleKind.NoModule =>
```

Comment:
You do not need this guard. `linker.linkUnit` is independent of the `ModuleKind`. It might be worth documenting this on `Linker`.